course_catalog/etl/video.py

```diff
@@ -0,0 +1,62 @@
+"""video catalog ETL"""
+
+from django.conf import settings
+from googleapiclient.discovery import build
+import yaml
+import requests
+
+YOUTUBE_API_SERVICE_NAME = "youtube"
+YOUTUBE_API_VERSION = "v3"
+
+
+def get_youtube_client():
+    """Client for google api"""
+    developer_key = settings.YT_DEVELOPER_KEY
+    return build(
+        YOUTUBE_API_SERVICE_NAME, YOUTUBE_API_VERSION, developerKey=developer_key
+    )
+
+
+def get_videos_for_channel_in_batches(youtube_client, channel_id, offered_by):
+    """Load youtube video results in batches"""
+    page_token = ""
+
+    while True:
+        search_request = youtube_client.search().list(
+            channelId=channel_id,
+            maxResults=50,
+            pageToken=page_token,
+            part="snippet",
+            type="video",
+        )
+
+        search_response = search_request.execute()
+
+        video_ids = map(lambda video: video["id"]["videoId"], search_response["items"])
+        video_ids_paramenter = ", ".join(video_ids)
+
+        full_request = youtube_client.videos().list(
+            part="snippet", id=video_ids_paramenter
+        )
+        full_response = full_request.execute()
+
+        yield (offered_by, full_response)
+
+        if "nextPageToken" in search_response:
+            page_token = search_response["nextPageToken"]
+        else:
+            break
+
+
+def extract():
+    """Extract for youtube data"""
+    youtube_client = get_youtube_client()
```

Comment:
Add a check at the beginning of this function that if either `VIDEOS_CONFIG_FILE_LOCATION` or `YT_DEVELOPER_KEY` aren't set it just returns (that will just cause the generator to yield nothing and it ends up as an empty list)